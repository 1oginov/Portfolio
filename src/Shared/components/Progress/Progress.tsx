import { CircularProgress } from '@material-ui/core';
import * as React from 'react';

interface Props {
  className?: string;
  size?: 'large';
}

const Progress: React.FunctionComponent<Props> = ({ className, size }: Props) => {
  let sizeProp;

  if (size === 'large') {
    sizeProp = 64;
  }

  return <CircularProgress className={className} size={sizeProp} />;
};

export default Progress;
