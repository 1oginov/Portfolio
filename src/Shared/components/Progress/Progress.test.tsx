/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import Progress from './Progress';

it('matches snapshot', () => {
  const wrapper = shallow(<Progress />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when size is large', () => {
  const wrapper = shallow(<Progress size="large" />);

  expect(wrapper).toMatchSnapshot();
});
