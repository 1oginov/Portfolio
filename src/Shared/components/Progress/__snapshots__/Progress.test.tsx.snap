// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `<WithStyles(ForwardRef(CircularProgress)) />`;

exports[`matches snapshot when size is large 1`] = `
<WithStyles(ForwardRef(CircularProgress))
  size={64}
/>
`;
